File: 剑指offer/03-从尾到头打印链表.py
# -*- coding:utf-8 -*-

"""
题目描述
输入一个链表，按链表值从尾到头的顺序返回一个ArrayList。
"""

# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # 返回从尾部到头部的列表值序列，例如[1,2,3]
    def printListFromTailToHead(self, listNode):
        # write code here
        arr = []
        if not listNode:
            return []
        else:
            arr.append(listNode.val)
        while listNode.next:
            listNode = listNode.next
            arr.append(listNode.val)
        return arr[::-1]

File: 剑指offer/03-从尾到头打印链表.js
// 题目描述
// 输入一个链表，按链表值从尾到头的顺序返回一个ArrayList。
/*function ListNode(x){
    this.val = x;
    this.next = null;
}*/
function printListFromTailToHead(head)
{
    // write code here
    var arr = [];
    if(head){ // 这里要注意牛客测试系统传过来的是null不是空对象
        arr.push(head.val);
    }else{
        return arr;
    }
    while(head.next){
        head = head.next;
        arr.push(head.val);
    }
    return arr.reverse();
}
module.exports = {
    printListFromTailToHead : printListFromTailToHead
};